### PR TITLE
chore: reduce log spam

### PR DESCRIPTION
**Pull request by tzneal**
Fixes # 

**Description**

**How was this change tested?**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/controllers/provisioning/provisioner.go [hunk 1]
```diff
@@ -214,23 +214,31 @@ func (p *Provisioner) GetPendingPods(ctx context.Context) ([]*v1.Pod, error) {
...
+
+// consolidationWarnings potentially writes logs warning about possible unexpected interactions between scheduling
+// constraints and consolidation
+func (p *Provisioner) consolidationWarnings(ctx context.Context, po v1.Pod) {
+	// We have pending pods that have preferred anti-affinity or topology spread constraints.  These can interact
+	// unexpectedly with consolidation so we warn once per hour when we see these pods.
+	if po.Spec.Affinity != nil && po.Spec.Affinity.PodAntiAffinity != nil {
+		if len(po.Spec.Affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution) != 0 {
+			if p.cm.HasChanged(string(po.UID), "pod-antiaffinity") {
+				logging.FromContext(ctx).Infof("pod %s has a preferred Anti-Affinity which can prevent consolidation", client.ObjectKeyFromObject(&po))
```
Comment (jonathan-innis) [inline pkg/controllers/provisioning/provisioner.go:231]: Should this be info or debug?

File: pkg/controllers/provisioning/provisioner.go [hunk 2]
```diff
@@ -214,23 +214,31 @@ func (p *Provisioner) GetPendingPods(ctx context.Context) ([]*v1.Pod, error) {
...
 			}
 		}
-		for _, tsc := range po.Spec.TopologySpreadConstraints {
-			if tsc.WhenUnsatisfiable == v1.ScheduleAnyway {
+	}
+	for _, tsc := range po.Spec.TopologySpreadConstraints {
+		if tsc.WhenUnsatisfiable == v1.ScheduleAnyway {
+			if p.cm.HasChanged(string(po.UID), "pod-topology-spread") {
 				logging.FromContext(ctx).Infof("pod %s has a preferred TopologySpreadConstraint which can prevent consolidation", client.ObjectKeyFromObject(&po))
```
Comment (jonathan-innis) [inline pkg/controllers/provisioning/provisioner.go:238]: Same question here.

File: pkg/controllers/provisioning/provisioner.go (hunk 1 above)
Comment (jonathan-innis) [inline pkg/controllers/provisioning/provisioner.go:231]:
```suggestion
				logging.FromContext(ctx).Infof("pod %s has a preferred antiAffinity which can prevent consolidation", client.ObjectKeyFromObject(&po))
```

nit: Prefer lowercase logging?

File: pkg/controllers/provisioning/provisioner.go (hunk 2 above)
Comment (jonathan-innis) [inline pkg/controllers/provisioning/provisioner.go:238]:
```suggestion
				logging.FromContext(ctx).Infof("pod %s has a preferred toplogySpreadConstraint which can prevent consolidation", client.ObjectKeyFromObject(&po))
```

nit: Prefer lowercase logging

**Review by jonathan-innis — APPROVED** [review]
Few comments/questions, otherwise LGTM

File: pkg/controllers/provisioning/provisioner.go (hunk 1 above)
Comment (ellistarn) [reply pkg/controllers/provisioning/provisioner.go:231]: Disagree w/ the recommendations -- I think @tzneal got it right here.. Proper nouns make sense to be capitalized. Starting each log line with a capital (as if a sentence), make less sense.

Comment (ellistarn) [reply pkg/controllers/provisioning/provisioner.go:231]: I suppose the camelCased version is the same as customers would write into their yaml.

Comment (ellistarn) [reply pkg/controllers/provisioning/provisioner.go:231]: > I'm not really sure what the proper "logging standards"

image

Comment (tzneal) [reply pkg/controllers/provisioning/provisioner.go:231]: Info on these at least to start, we're trying to provide more visibility as this behavior has been unexpected by a few customers.

**Pull request merged**